### ISSUE TITLE
Correct (?) a few reference screenshots 

### ISSUE THREAD

**Issue by emilyrohrbough** [cerner/terra-framework#711]
# Investigation Needed

The recent updates made to terra-core and terra-framework mono-repos to consume terra-toolkit v5 and terra-dev-site v5 exposed a few discrepancies in the _wdio_ reference screenshots for the components listed below. 

**terra-abstract-modal**
📝 Focus seems to have changed - button in the modal seems to have lost focus. This change looks legit but need a confirmation so that the theme repo can update them too.

[1] https://github.com/cerner/terra-framework/pull/685/files?short_path=8c9a696#diff-8c9a6961ccebb2c5460fe94fd74b7b74
[2] https://github.com/cerner/terra-framework/pull/685/files?short_path=98b74c2#diff-a352c6af1b344ea67d528d112d9b1983
[3] https://github.com/cerner/terra-framework/pull/685/files?short_path=98b74c2#diff-5be4a9946d4ed66e68fb023e1a264072

### Mentions
@nramamurth

**Comment by bjankord** [comment]
When I looked at the linked screenshots, number 1 looks odd, but 2 and 3 look correct to me.

These 2 screenshots have a noticeable diff, the buttons used to have a focus outline but now they don't. These were the only screenshots in abstract modal that looked odd to me.
* [packages/terra-abstract-modal/tests/wdio/snapshots/reference/en/chrome_medium/abstract-modal-spec/Is_Fullscreen[modal_closed,_with_focus].png](https://github.com/cerner/terra-framework/pull/685/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.jsx&file-filters%5B%5D=.md&file-filters%5B%5D=.png&file-filters%5B%5D=.snap&file-filters%5B%5D=dotfile&owned-by%5B%5D=#diff-8c9a6961ccebb2c5460fe94fd74b7b74)
* [packages/terra-abstract-modal/tests/wdio/snapshots/reference/en/chrome_medium/abstract-modal-spec/Is_Fullscreen[modal_closed,_with_focus].png](https://github.com/cerner/terra-framework/pull/685/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.jsx&file-filters%5B%5D=.md&file-filters%5B%5D=.png&file-filters%5B%5D=.snap&file-filters%5B%5D=dotfile&owned-by%5B%5D=&short_path=a352c6a#diff-a352c6af1b344ea67d528d112d9b1983)

The first one is the same 1 that @nramamurth pointed out.

I ran the steps from the wdio test for these tests manually in the browser at the test URL and they function correctly. I confirmed the button that toggles the modal open is focused by checking `document.activeElement` in the console after the steps were complete and the button is listed as the document.activeElement. However, I did notice that no focus outline was added to the button in Chrome.

* [modal trigger focus test when closing the modal](https://github.com/cerner/terra-framework/blob/master/packages/terra-abstract-modal/tests/wdio/abstract-modal-spec.js#L124-L138)
* [fullscreen modal trigger focus test when closing the modal](https://github.com/cerner/terra-framework/blob/master/packages/terra-abstract-modal/tests/wdio/abstract-modal-spec.js#L102-L122)

I found it odd that the native button didn't have a focus outline when we shift focus back to it. I tested using the dev tools to trigger the focus styles on the native button and saw that there was no focus outline applied to the native button when the focus state was triggered from the dev tools.

<img width="760" alt="Screen Shot 2019-06-07 at 9 24 42 PM" src="https://user-images.githubusercontent.com/633148/59140989-eef3bb00-896a-11e9-9077-4e137ce1ab9e.png">

tl;dr I think all the abstract modal screenshots are correct, but the version of chrome we now use and the version of chrome on my local machine seem to not show focus outlines on the native button when we shift focus to it programmatically in this scenario.

I think we should make a demo button in terra-doc-template that has clear focus styles that may be more helpful to see when a button we use for testing purposes actually has focus.

**Comment by bjankord** [comment]
There is a test in the layout-slide-panel package that looks to have the same issue, we programmatically shift focus to a native button but the native button no longer shows a focus outline: https://github.com/cerner/terra-framework/pull/685/files#diff-5c83d56386dbb236055469e4c5d86c17

I checked document.activeElement after completing the test steps for this test in the browser manually and the button is receiving focus correctly.

**Comment by bjankord** [comment]
These tests look to have a similar issue where clicking on a native anchor link used to render a focus outline and now it doesn't:

* https://github.com/cerner/terra-framework/pull/685/files#diff-366797327db87a3e597ba5b08d951e8c
* https://github.com/cerner/terra-framework/pull/685/files#diff-b97352a2096f7f6e7dfe5cd6fe439ee8
* https://github.com/cerner/terra-framework/pull/685/files#diff-5d942486a02160ee7807340530f86503

I'm not entirely sure what is triggering the focus outline to no longer appear but I think it is likely a chrome bug of some sort.

On these images, it looks like we are capturing the screenshot before the route changes, we should likely add a wait statement for content in the 2nd route to load before taking a screenshot.
* https://github.com/cerner/terra-framework/pull/685/files#diff-262d7f39d3f5dfce0342c8f94d9b2fe0
* https://github.com/cerner/terra-framework/pull/685/files#diff-b7915edd77cf97a4785636e610c8f7dc

**Comment by emilyrohrbough** [comment]
@bjankord Is there a bug in how we are setting the focus back to the previously focused button if it has focus but it is not correctly receiving the styles? This sounds like it is a similar issue the terra-form-select is having: https://github.com/cerner/terra-core/issues/2421

**Comment by bjankord** [comment]
@emilyrohrbough The button is receiving focus when we close the modal when the button is used to pen the modal. This can be checked by viewing `document.activeElement` in browser console after closing the modal.

In scrum today we've chalked this up as Chrome no longer presenting focus outlines if the button is not focused by keyboard. Plan to remove the screenshot tests and replace with a hasFocus test.
